src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,157 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidUserDataException;
+import core.basesyntax.exception.UserExistsException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private User testUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.add(new User("login1", "password1", 18));
+        testUser = new User("validLogin", "validPassword", 28);
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_userIsNull_notOk() {
+        testUser = null;
+        assertThrows(IllegalArgumentException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_userAlreadyExists_notOk() {
+        testUser.setLogin("login1");
+        assertThrows(UserExistsException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_userLoginIsNull_notOk() {
+        testUser.setLogin(null);
+        assertThrows(IllegalStateException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_userLoginIsEmpty_notOk() {
+        testUser.setLogin("");
+        assertThrows(InvalidUserDataException.class, () -> registrationService.register(testUser));
+    }
+
+    @Test
+    void register_userLoginUnderSixSymbols_notOk() {
```

Comment:
do not hardcode values in the names
```suggestion
    void register_shortUserLogin_notOk() {
```